### PR TITLE
28 add set default rich menu to line bot

### PR DESCRIPTION
**Pull request by sarge963**
Added SetDefaultRichMenu to LineBot according to issue #28 Add SetDefaultRichMenu to LineBot

As usually, all comments are welcomed. 😉

### CODE REVIEW

File: src/LineBot/LineBot.cs [hunk 1]
```diff
@@ -489,6 +489,37 @@ public async Task<ILineBot> Reply(string replyToken, params ISendMessage[] messa
...
+        /// Sets the default rich menu, which is displayed to all users who have added your bot as a friend and are not linked to any per-user rich menu.
+        /// </summary>
+        /// <param name="richMenuId">The rich menu id.</param>
+        /// <returns>.</returns>
+        public async Task<ILineBot> SetDefaultRichMenu(string richMenuId)
+        {
+            Guard.NotNullOrEmpty(nameof(richMenuId), richMenuId);
+
+            var emptyObject = new object();
+            var content = CreateStringContent(emptyObject);
```
Comment (dlemstra) [inline src/LineBot/LineBot.cs]: Wondering if we could not just set content to `{}` instead?

File: tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+namespace Line.Tests
+{
+    public partial class LineBotTests
+    {
+        [TestClass]
+        public class TheSetDefaultRichMenuMethod
+        {
+            [TestMethod]
+            public async Task ThrowsExceptionWhenRichMenuIdIsNull()
```
Comment (dlemstra) [inline tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs]: Could you rename the methods to start with `Should` instead? The newer unit test use a different naming convention.

File: tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            {
+                ILineBot bot = TestConfiguration.CreateBot();
```
Comment (dlemstra) [inline tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs]: Could you use `var` where possible?

File: tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Line.Tests
+{
+    public partial class LineBotTests
+    {
+        [TestClass]
+        public class TheSetDefaultRichMenuMethod
```
Comment (dlemstra) [inline tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs]: We probably also need a unit test that verifies that the current instance is returned. This should probably also be added or a lot of other tests. Could you do this in a new PR instead?

File: src/LineBot/LineBot.cs (hunk 1 above)
Comment (sarge963) [reply src/LineBot/LineBot.cs]: Current implementation creates content as {}. 
The CreateStringContent("{}"); will create content as "{}". 

I can simplify 2 lines into 1 like this: var content = CreateStringContent(new object());
What do you prefer?

Comment (dlemstra) [reply src/LineBot/LineBot.cs]: Just took another look at the API and it seems that you should not post an object? Maybe pass `null` to `PostAsync` instead?

**Comment by sarge963** [comment]
Any further comments. 😉

**Comment by sarge963** [comment]
Done.

File: tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                public async Task ShouldThrowExceptionWhenRichMenuIdIsEmpty()
+                {
+                    var richMenuResponse = new RichMenuResponse()
+                    {
+                        Id = string.Empty
+                    };
+
+                    var bot = TestConfiguration.CreateBot();
+
+                    await ExceptionAssert.ThrowsArgumentException("richMenuId", "Value cannot be empty.", async () =>
```
Comment (dlemstra) [inline tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs]: And here...

File: tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                        await bot.SetDefaultRichMenu(null);
+                    });
+                }
+
+                [TestMethod]
+                public async Task ShouldThrowExceptionWhenRichMenuIdIsEmpty()
+                {
+                    var bot = TestConfiguration.CreateBot();
+
+                    await ExceptionAssert.ThrowsArgumentException("richMenuId", "Value cannot be empty.", async () =>
```
Comment (dlemstra) [inline tests/LineBot.Tests/LineBotTests/TheSetDefaultRichMenuMethod.cs]: You can use `ThrowsArgumentEmptyExceptionAsync` here instead.

**Comment by sarge963** [comment]
Done. :)

File: tests/LineBot.Tests/TestHelpers/ExceptionAssert.cs [hunk 7]
```diff
@@ -73,6 +73,12 @@ public static void ThrowsArgumentException(string paramName, string message, Act
             Assert.AreEqual(paramName, exception.ParamName);
         }
 
+        public static async Task ThrowsArgumentException(string paramName, string message, Func<Task> action)
```
Comment (dlemstra) [inline tests/LineBot.Tests/TestHelpers/ExceptionAssert.cs]: Can you remove this because we no longer use it?

**Comment by sarge963** [comment]
No problem, I just need to get rid of my bad habits 🙂

**Pull request merged**